server.js

```diff
@@ -575,6 +575,59 @@ Server.prototype._onWebSocketRequest = function (socket, opts, params) {
   })
 }
 
+Server.prototype._onWebSocketSend = function (socket, err) {
+  var self = this
+  if (err) self._onWebSocketError(socket, err)
+}
+
+Server.prototype._onWebSocketClose = function (socket) {
+  var self = this
+  debug('websocket close %s', socket.peerId)
+
+  if (socket.peerId) {
+    socket.infoHashes.slice(0).forEach(function (infoHash) {
```

Comment:
We need to clone the array before looping over it, otherwise it behaves incorrectly because we're mutating it inside the loop here: https://github.com/feross/bittorrent-tracker/pull/198/files/ebe3c218feab4f5a6774d5092531c6b2e55fe608#diff-831f46884145ede920506b29d5681ab2R104